File: scripts/igaming-gemini.py
#!/usr/bin/env python3
"""
Sistema Unificado iGaming - Gemini 3 Pro Image API
Usa a API do Gemini diretamente (sem Replicate)
Suporta até 5 imagens de referência para manter consistência do personagem
"""
import sys
import os
import json
import time
import base64
import random
import argparse
from datetime import datetime
from pathlib import Path

# Instalar dependências se necessário
try:
    from google import genai
    from google.genai import types
    from PIL import Image
except ImportError:
    print("Instalando dependências...")
    os.system("pip install google-genai pillow --quiet")
    from google import genai
    from google.genai import types
    from PIL import Image

try:
    from googleapiclient.discovery import build
    from googleapiclient.http import MediaFileUpload
    from google.oauth2 import service_account
    from google.oauth2.credentials import Credentials
    from google.auth.transport.requests import Request
except ImportError:
    os.system("pip install google-api-python-client --quiet")
    from googleapiclient.discovery import build
    from googleapiclient.http import MediaFileUpload
    from google.oauth2 import service_account
    from google.oauth2.credentials import Credentials
    from google.auth.transport.requests import Request

# Configurações
GEMINI_API_KEY = os.getenv('GEMINI_API_KEY', 'AIzaSyAerWKegKaAUh5idI-Ra0sjEXcSSTXkp90')
SCOPES = ['https://www.googleapis.com/auth/drive.file']
SERVICE_ACCOUNT_FILE = 'C:\\Users\\vsuga\\clawd\\config\\imperio-service-account.json'
TOKEN_FILE = 'C:\\Users\\vsuga\\clawd\\config\\token.json'
RCLONE_PATH = 'C:\\Users\\vsuga\\clawd\\rclone.exe'
OUTPUT_DIR = Path('C:\\Users\\vsuga\\clawd\\images\\generated')
OUTPUT_DIR.mkdir(parents=True, exist_ok=True)

# Perfis iGaming
PROFILES = {
    "jonathan": {
        "name": "Jhonatan Vieira",
        "folderId": "1-pRp7UtxfBVBNw1-5WJPCtzF5PnTmNUZ",
        "referencePhoto": "jonathan.jpg",
        "characteristics": "Homem brasileiro, cabelo curto escuro, barba, pele morena, expressão confiante"
    },
    "pedro": {
        "name": "Pedro vlEIRA", 
        "folderId": "16Mhy_ydDXeq2RuvWq3F1FQ9Ehei5tsa7",
        "referencePhoto": "pedro.jpg",
        "characteristics": "Homem brasileiro, careca, barba, expressão confiante e determinada"
    },
    "laise": {
        "name": "Laise",
        "folderId": "18vm4Fv1hYM8B89m-qhr-eUeZjxKmm9Zm",
        "referencePhoto": "laise.jpg",
        "characteristics": "Mulher brasileira, cabelo longo, sorriso natural, elegante"
    },
    "teo": {
        "name": "Teo Martins",
        "folderId": "1jDY5HSjJOtZw2yCxXzuEy_gTt9uRTObP",
        "referencePhoto": "teo.jpg",
        "characteristics": "Homem brasileiro, barba escura aparada, cabelo preto, olhos claros, pele morena, carismático"
    },
    "fernandes": {
        "name": "Fernandes Vieira",
        "folderId": None,
        "referencePhoto": "fernandes.jpg",
        "characteristics": "Homem brasileiro, cabelo curto, expressão profissional"
    }
}

# Cenários Lifestyle Realistas
SCENARIOS = {
    "viagem": [
        "em frente ao Coliseu em Roma, dia ensolarado, vestindo roupa casual elegante",
        "na Torre Eiffel à noite, luzes brilhantes de Paris ao fundo",
        "em Santorini, Grécia, casas brancas e mar azul turquesa ao fundo",
        "em Dubai, Burj Khalifa ao fundo, cidade moderna luxuosa",
        "em Nova York, Times Square, energia urbana vibrante",
        "nas Maldivas, overwater bungalow, águas cristalinas",
        "em Tóquio, ruas com neon, estilo moderno",
        "no Grand Canyon, paisagem épica, aventura ao ar livre"
    ],
    "praia": [
        "em praia paradisíaca no Caribe, areia branca, água cristalina, bermuda de banho",
        "em resort de luxo, piscina infinity com vista pro mar",
        "em beach club, ambiente festivo e sofisticado",
        "em iate no mar, deck espaçoso, vida luxuosa",
        "em praia brasileira, coqueiros, céu azul perfeito"
    ],
    "academia": [
        "em academia moderna de luxo, equipamentos high-tech, roupa esportiva",
        "fazendo crossfit, box moderno, iluminação industrial",
        "levantando peso, músculos definidos, foco total",
        "em aula de boxing, luvas de boxe, determinação"
    ],
    "restaurante": [
        "em restaurante gourmet sofisticado, iluminação íntima, terno elegante",
        "em rooftop bar com vista para cidade à noite, drink na mão",
        "em wine bar exclusivo, taças de vinho, ambiente romântico",
        "em brunch em café parisiense, croissant e café"
    ],
    "trabalho": [
        "em escritório executivo moderno, vista panorâmica da cidade, terno",
        "em coworking tech, ambiente criativo e inovador",
        "em reunião de negócios, apresentando para investidores",
        "em home office elegante, setup profissional"
    ],
    "lifestyle": [
        "dirigindo carro de luxo conversível na estrada costeira",
        "em jato privado, interior luxuoso de couro",
        "em festa exclusiva, ambiente VIP, roupa elegante",
        "comprando em shopping de luxo, sacolas de grife"
    ],
    "cassino": [
        "em cassino luxuoso de Las Vegas, luzes neon vibrantes, terno elegante",
        "em mesa de poker VIP, fichas empilhadas, ambiente exclusivo",
        "em sala de roleta, ambiente sofisticado e tenso"
    ]
}

# Copys iGaming
COPYS = [
    "QUAL O SEU BOLETO MAIS CARO? DEIXA AQUI NOS COMENTÁRIOS 👇🔥",
    "Manda aqui e concorra 🎁🔥",
    "Participa aqui 🫡💰",
    "Manda que eu to sorteando 🎁😎",
    "Comenta aqui e ganha 🔥🫡",
    "Manda aqui pro prêmio 💰👇",
    "Quer saber como eu faço? Comenta EU QUERO 👇",
    "Qual foi sua maior vitória? 🏆💰",
    "Quem quer participar? Deixa o 🔥 nos comentários",
    "Isso aqui é real, comenta pra participar 💰",
    "Manda a reação e participa 🎲✨",
    "Comenta e concorre 😎💰",
    "Próximo ganhador pode ser você 🎁💰",
    "Manda que eu to de olho 👀🫡",
    "Participa do prêmio 🫡😎",
    "Manda aqui e boa sorte 🍀💰"
]

def get_oauth_drive():
    """Conecta ao Drive via OAuth"""
    creds = None
    if os.path.exists(TOKEN_FILE):
        creds = Credentials.from_authorized_user_file(TOKEN_FILE, SCOPES)
    if not creds or not creds.valid:
        if creds and creds.expired and creds.refresh_token:
            creds.refresh(Request())
    return build('drive', 'v3', credentials=creds) if creds else None

def get_reference_image(profile_name):
    """Busca imagem de referência do perfil"""
    profile = PROFILES.get(profile_name)
    if not profile:
        return None
    
    # Tentar vários caminhos
    paths = [
        Path(f"C:\\Users\\vsuga\\clawd\\{profile['referencePhoto']}"),
        Path(f"C:\\Users\\vsuga\\clawd\\images\\{profile['referencePhoto']}"),
        Path(f".\\{profile['referencePhoto']}")
    ]
    
    for ref_path in paths:
        if ref_path.exists():
            return ref_path
    
    return None

def generate_image_gemini(ref_image_path, prompt, aspect_ratio="1:1", resolution="2K"):
    """Gera imagem usando Gemini 3 Pro Image API diretamente"""
    print(f"   🎨 Gerando com Gemini 3 Pro Image...")
    print(f"   📐 Aspecto: {aspect_ratio} | Resolução: {resolution}")
    
    try:
        # Inicializar cliente
        client = genai.Client(api_key=GEMINI_API_KEY)
        
        # Carregar imagem de referência
        ref_image = Image.open(ref_image_path)
        
        # Gerar
        response = client.models.generate_content(
            model="gemini-3-pro-image-preview",
            contents=[
                prompt,
                ref_image  # Imagem de referência para manter consistência
            ],
            config=types.GenerateContentConfig(
                response_modalities=['TEXT', 'IMAGE'],
                image_config=types.ImageConfig(
                    aspect_ratio=aspect_ratio,
                    image_size=resolution
                )
            )
        )
        
        # Extrair imagem da resposta
        for part in response.parts:
            if hasattr(part, 'as_image') and (image := part.as_image()):
                print(f"   ✅ Imagem gerada com sucesso!")
                return image
            elif hasattr(part, 'text') and part.text:
                print(f"   📝 Resposta: {part.text[:100]}...")
        
        print("   ❌ Nenhuma imagem na resposta")
        return None
        
    except Exception as e:
        print(f"   ❌ Erro: {e}")
        import traceback
        traceback.print_exc()
        return None

def upload_to_drive(image, folder_id, filename, service):
    """Upload imagem para o Drive"""
    local_path = OUTPUT_DIR / filename
    
    # Salvar localmente
    image.save(local_path)
    print(f"   💾 Salvo: {local_path}")
    
    if service and folder_id:
        try:
            media = MediaFileUpload(str(local_path), mimetype='image/png')
            file = service.files().create(
                body={'name': filename, 'parents': [folder_id]},
                media_body=media,
                fields='id, webViewLink'
            ).execute()
            print(f"   ☁️ Upload: {file.get('webViewLink')}")
            return {'path': str(local_path), 'link': file.get('webViewLink')}
        except Exception as e:
            print(f"   ⚠️ Erro upload: {e}")
    
    # Fallback: usar RClone
    if folder_id:
        target = f"gdrive:{folder_id}/{filename}"
        os.system(f'"{RCLONE_PATH}" copyto "{local_path}" "{target}" 2>nul')
        print(f"   ☁️ Upload via RClone")
    
    return {'path': str(local_path)}

def generate_for_profile(profile_name, category=None, count=1, aspect="1:1", resolution="2K"):
    """Gera imagens para um perfil específico"""
    profile = PROFILES.get(profile_name)
    if not profile:
        print(f"❌ Perfil '{profile_name}' não encontrado")
        print(f"   Disponíveis: {', '.join(PROFILES.keys())}")
        return []
    
    if not profile['folderId']:
        print(f"⚠️ Perfil '{profile_name}' sem folder ID - salvando apenas localmente")
    
    print(f"\n🎨 Gerando para: {profile['name']}")
    print(f"📁 Pasta: {profile['folderId'] or 'LOCAL'}")
    
    # Buscar imagem de referência
    ref_path = get_reference_image(profile_name)
    if not ref_path:
        print(f"❌ Imagem de referência não encontrada: {profile['referencePhoto']}")
        print(f"   Coloque o arquivo em: C:\\Users\\vsuga\\clawd\\{profile['referencePhoto']}")
        return []
    
    print(f"✅ Referência: {ref_path}")
    
    # Escolher cenários
    if category and category in SCENARIOS:
        scenarios = SCENARIOS[category]
        selected = random.sample(scenarios, min(count, len(scenarios)))
    else:
        # Mix de todas as categorias
        all_scenarios = []
        for cat_scenarios in SCENARIOS.values():
            all_scenarios.extend(cat_scenarios)
        selected = random.sample(all_scenarios, min(count, len(all_scenarios)))
    
    # Conectar Drive
    try:
        drive = get_oauth_drive()
    except Exception as e:
        print(f"⚠️ Sem conexão OAuth: {e}")
        drive = None
    
    generated = []
    
    for i, scenario in enumerate(selected):
        print(f"\n[{i+1}/{count}] 🎬 Cenário: {scenario[:60]}...")
        
        # Montar prompt completo
        prompt = f"""Generate a realistic photograph of this EXACT person: {profile['characteristics']}.

Scene: {scenario}

CRITICAL: Maintain the EXACT same face, features, and identity from the reference image.
Style: Professional Instagram photography, natural lighting, high quality, realistic.
"""
        
        # Gerar imagem
        image = generate_image_gemini(ref_path, prompt, aspect, resolution)
        
        if image:
            ts = datetime.now().strftime("%Y%m%d_%H%M%S")
            cat_name = category or 'lifestyle'
            filename = f"{profile_name}_{cat_name}_{ts}.png"
            
            # Upload
            result = upload_to_drive(image, profile['folderId'], filename, drive)
            
            copy = random.choice(COPYS)
            generated.append({
                'filename': filename,
                'scenario': scenario,
                'copy': copy,
                'path': result.get('path'),
                'link': result.get('link')
            })
        else:
            print(f"   ❌ Falha na geração")
        
        # Pausa entre gerações (rate limiting)
        if i < len(selected) - 1:
            print("   ⏳ Aguardando 3s...")
            time.sleep(3)
    
    # Resumo
    print(f"\n{'='*50}")
    print(f"🎉 GERAÇÃO COMPLETA: {len(generated)}/{count}")
    print(f"{'='*50}")
    
    for g in generated:
        print(f"\n📷 {g['filename']}")
        print(f"   📍 {g['scenario'][:50]}...")
        print(f"   📝 Copy: {g['copy']}")
        if g.get('link'):
            print(f"   🔗 {g['link']}")
    
    return generated

def main():
    parser = argparse.ArgumentParser(
        description='Sistema iGaming - Gemini 3 Pro Image',
        formatter_class=argparse.RawDescriptionHelpFormatter,
        epilog="""
Exemplos:
  python igaming-gemini.py -p teo -c praia -n 3
  python igaming-gemini.py -p jonathan -c viagem -n 5 --aspect 16:9
  python igaming-gemini.py -p laise -n 2 --resolution 4K
        """
    )
    parser.add_argument('--profile', '-p', required=True, 
                       choices=list(PROFILES.keys()),
                       help='Perfil: jonathan, pedro, laise, teo, fernandes')
    parser.add_argument('--category', '-c', 
                       choices=list(SCENARIOS.keys()),
                       help='Categoria: viagem, praia, academia, restaurante, trabalho, lifestyle, cassino')
    parser.add_argument('--count', '-n', type=int, default=1,
                       help='Número de imagens (default: 1)')
    parser.add_argument('--aspect', '-a', default='1:1',
                       choices=['1:1', '2:3', '3:2', '3:4', '4:3', '4:5', '5:4', '9:16', '16:9', '21:9'],
                       help='Proporção (default: 1:1)')
    parser.add_argument('--resolution', '-r', default='2K',
                       choices=['1K', '2K', '4K'],
                       help='Resolução (default: 2K)')
    
    args = parser.parse_args()
    
    print("🎮 SISTEMA IGAMING - GEMINI 3 PRO IMAGE")
    print("=" * 50)
    print(f"🔑 API Key: ...{GEMINI_API_KEY[-8:]}")
    
    generate_for_profile(
        args.profile, 
        args.category, 
        args.count,
        args.aspect,
        args.resolution
    )

if __name__ == "__main__":
    main()

File: scripts/igaming-gemini.js
/**
 * Sistema iGaming - Gemini 3 Pro Image
 * 
 * Fluxo:
 * 1. Pega foto de referência do perfil (pasta do Drive)
 * 2. Gera nova imagem com Gemini usando a referência + cenário
 * 3. Aplica copy overlay na imagem
 * 4. Salva na pasta do perfil no Drive
 * 
 * Uso: node igaming-gemini.js -p teo -c praia -n 3
 */

const https = require('https');
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

// Configurações
const GEMINI_API_KEY = 'AIzaSyAerWKegKaAUh5idI-Ra0sjEXcSSTXkp90';
const RCLONE_PATH = 'C:\\Users\\vsuga\\clawd\\rclone.exe';
const OUTPUT_DIR = 'C:\\Users\\vsuga\\clawd\\images\\generated';
const REFERENCES_DIR = 'C:\\Users\\vsuga\\clawd\\images\\references';

// Criar pastas
[OUTPUT_DIR, REFERENCES_DIR].forEach(dir => {
    if (!fs.existsSync(dir)) fs.mkdirSync(dir, { recursive: true });
});

// Pasta de estilo/copys de referência no Drive
const STYLE_FOLDER_ID = "19w3WefIuH18POsomvRpEhAZZzyrjqj50";

// Perfis iGaming - cada um com sua pasta de fotos no Drive
const PROFILES = {
    teo: {
        name: "Teo Martins",
        folderId: "1jDY5HSjJOtZw2yCxXzuEy_gTt9uRTObP",
        characteristics: "Homem brasileiro, barba escura aparada, cabelo preto, olhos claros, pele morena, carismático, expressão confiante"
    },
    jonathan: {
        name: "Jhonatan Vieira",
        folderId: "1-pRp7UtxfBVBNw1-5WJPCtzF5PnTmNUZ",
        characteristics: "Homem brasileiro, cabelo curto escuro, barba, pele morena, expressão confiante"
    },
    pedro: {
        name: "Pedro vlEIRA",
        folderId: "16Mhy_ydDXeq2RuvWq3F1FQ9Ehei5tsa7",
        characteristics: "Homem brasileiro, careca, barba, expressão confiante e determinada"
    },
    laise: {
        name: "Laise",
        folderId: "18vm4Fv1hYM8B89m-qhr-eUeZjxKmm9Zm",
        characteristics: "Mulher brasileira, cabelo longo, sorriso natural, elegante"
    },
    fernandes: {
        name: "Fernandes Vieira",
        folderId: null,
        characteristics: "Homem brasileiro, cabelo curto, expressão profissional"
    }
};

// Cenários Lifestyle
const SCENARIOS = {
    viagem: [
        "em frente ao Coliseu em Roma, dia ensolarado",
        "na Torre Eiffel à noite, luzes brilhantes de Paris",
        "em Santorini, Grécia, casas brancas e mar azul",
        "em Dubai, Burj Khalifa ao fundo",
        "em Nova York, Times Square",
        "nas Maldivas, águas cristalinas"
    ],
    praia: [
        "em praia paradisíaca, areia branca, água cristalina",
        "em resort de luxo, piscina infinity",
        "em beach club sofisticado",
        "em iate no mar"
    ],
    academia: [
        "em academia moderna de luxo",
        "fazendo crossfit, box moderno",
        "levantando peso, foco total"
    ],
    restaurante: [
        "em restaurante gourmet sofisticado",
        "em rooftop bar com vista para cidade",
        "em wine bar exclusivo"
    ],
    trabalho: [
        "em escritório executivo moderno",
        "em coworking tech",
        "em reunião de negócios"
    ],
    lifestyle: [
        "dirigindo carro de luxo",
        "em jato privado",
        "em festa exclusiva VIP"
    ],
    cassino: [
        "em cassino luxuoso de Las Vegas",
        "em mesa de poker VIP",
        "em sala de roleta sofisticada"
    ]
};

// Copy iGaming
const COPY = "🔥 *Manda aqui 🔥👇🏻*";

function getRandomItem(arr) {
    return arr[Math.floor(Math.random() * arr.length)];
}

// Baixa uma foto de referência da pasta do perfil no Drive
function fetchReferenceFromProfileFolder(profileName) {
    const profile = PROFILES[profileName];
    if (!profile || !profile.folderId) return null;

    const localPath = path.join(REFERENCES_DIR, `${profileName}_ref.png`);

    // Usar cache se existir (menos de 1 dia)
    if (fs.existsSync(localPath)) {
        const stats = fs.statSync(localPath);
        const ageMs = Date.now() - stats.mtimeMs;
        if (ageMs < 24 * 60 * 60 * 1000) {
            console.log(`   📁 Usando referência em cache: ${localPath}`);
            return localPath;
        }
    }

    console.log(`   ⬇️ Buscando foto de referência do Drive...`);

    try {
        // Listar arquivos da pasta do perfil
        const files = execSync(
            `"${RCLONE_PATH}" lsf "gdrive:${profile.folderId}" --files-only`,
            { encoding: 'utf-8' }
        ).trim().split('\n').filter(f => f.endsWith('.png') || f.endsWith('.jpg'));

        if (files.length === 0) {
            console.log(`   ❌ Nenhuma imagem na pasta`);
            return null;
        }

        // Escolher uma imagem aleatória como referência
        const refFile = getRandomItem(files);
        console.log(`   📷 Baixando: ${refFile}`);

        const source = `gdrive:${profile.folderId}/${refFile}`;
        execSync(`"${RCLONE_PATH}" copyto "${source}" "${localPath}"`, { stdio: 'pipe' });

        if (fs.existsSync(localPath)) {
            console.log(`   ✅ Referência baixada!`);
            return localPath;
        }
    } catch (e) {
        console.log(`   ❌ Erro: ${e.message}`);
    }

    return null;
}

// Gera imagem com Gemini 3 Pro Image
async function generateWithGemini(refImagePath, prompt, aspectRatio = '1:1', resolution = '2K') {
    console.log('   🎨 Gerando com Gemini 3 Pro Image...');

    return new Promise((resolve) => {
        const imageBuffer = fs.readFileSync(refImagePath);
        const imageBase64 = imageBuffer.toString('base64');
        const mimeType = refImagePath.endsWith('.png') ? 'image/png' : 'image/jpeg';

        const payload = {
            contents: [{
                parts: [
                    { text: prompt },
                    { inline_data: { mime_type: mimeType, data: imageBase64 } }
                ]
            }],
            generationConfig: {
                responseModalities: ["TEXT", "IMAGE"],
                imageConfig: {
                    aspectRatio: aspectRatio,
                    imageSize: resolution
                }
            }
        };

        const postData = JSON.stringify(payload);

        const options = {
            hostname: 'generativelanguage.googleapis.com',
            path: `/v1beta/models/gemini-3-pro-image-preview:generateContent?key=${GEMINI_API_KEY}`,
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Content-Length': Buffer.byteLength(postData)
            }
        };

        const req = https.request(options, (res) => {
            let body = '';
            res.on('data', chunk => body += chunk);
            res.on('end', () => {
                try {
                    const response = JSON.parse(body);

                    if (response.error) {
                        console.log(`   ❌ Erro API: ${response.error.message}`);
                        resolve(null);
                        return;
                    }

                    if (response.candidates?.[0]?.content?.parts) {
                        for (const part of response.candidates[0].content.parts) {
                            if (part.inlineData?.data) {
                                console.log('   ✅ Imagem gerada!');
                                resolve(Buffer.from(part.inlineData.data, 'base64'));
                                return;
                            }
                        }
                    }

                    console.log('   ❌ Sem imagem na resposta');
                    resolve(null);
                } catch (e) {
                    console.log(`   ❌ Erro: ${e.message}`);
                    resolve(null);
                }
            });
        });

        req.on('error', (e) => {
            console.log(`   ❌ Erro: ${e.message}`);
            resolve(null);
        });

        req.write(postData);
        req.end();
    });
}

// Upload para Drive
function uploadToDrive(localPath, folderId, filename) {
    if (!folderId) return false;

    try {
        const target = `gdrive:${folderId}/${filename}`;
        execSync(`"${RCLONE_PATH}" copyto "${localPath}" "${target}"`, { stdio: 'pipe' });
        console.log('   ☁️ Upload concluído!');
        return true;
    } catch (e) {
        console.log(`   ⚠️ Erro upload: ${e.message}`);
        return false;
    }
}

// Função principal
async function generateForProfile(profileName, category, count, aspectRatio, resolution) {
    const profile = PROFILES[profileName];
    if (!profile) {
        console.log(`❌ Perfil '${profileName}' não encontrado`);
        return [];
    }

    console.log(`\n🎨 Gerando para: ${profile.name}`);
    console.log(`📁 Pasta: ${profile.folderId || 'LOCAL'}`);

    // Buscar foto de referência da pasta do perfil
    const refPath = fetchReferenceFromProfileFolder(profileName);
    if (!refPath) {
        console.log(`❌ Não foi possível obter referência`);
        return [];
    }

    // Escolher cenários
    let scenarios;
    if (category && SCENARIOS[category]) {
        scenarios = [...SCENARIOS[category]];
    } else {
        scenarios = Object.values(SCENARIOS).flat();
    }

    const shuffled = scenarios.sort(() => Math.random() - 0.5);
    const selected = shuffled.slice(0, count);

    const generated = [];

    for (let i = 0; i < selected.length; i++) {
        const scenario = selected[i];
        const copy = COPY;

        console.log(`\n[${i + 1}/${count}] 🎬 ${scenario.substring(0, 50)}...`);

        // Prompt que inclui instrução para colocar a copy NA imagem
        const prompt = `Generate a realistic photograph of this EXACT person: ${profile.characteristics}.

Scene: ${scenario}

TEXT OVERLAY: Add the following text prominently on the image in a stylish, Instagram-worthy way:
"${copy}"

IMPORTANT:
- Maintain the EXACT same face from the reference
- The text should be readable and well-positioned
- Use modern typography, maybe with a slight shadow or glow
- Professional Instagram photo quality`;

        const imageBuffer = await generateWithGemini(refPath, prompt, aspectRatio, resolution);

        if (imageBuffer) {
            const ts = new Date().toISOString().replace(/[-:T]/g, '').substring(0, 14);
            const catName = category || 'lifestyle';
            const filename = `${profileName.toUpperCase()}_${catName.toUpperCase()}_${ts}.png`;
            const localPath = path.join(OUTPUT_DIR, filename);

            fs.writeFileSync(localPath, imageBuffer);
            console.log(`   💾 Salvo: ${filename}`);

            uploadToDrive(localPath, profile.folderId, filename);

            generated.push({ filename, scenario, copy, path: localPath });
        } else {
            console.log('   ❌ Falha');
        }

        if (i < selected.length - 1) {
            console.log('   ⏳ Aguardando 3s...');
            await new Promise(r => setTimeout(r, 3000));
        }
    }

    // Resumo
    console.log(`\n${'='.repeat(50)}`);
    console.log(`🎉 COMPLETO: ${generated.length}/${count}`);
    console.log(`${'='.repeat(50)}`);

    for (const g of generated) {
        console.log(`\n📷 ${g.filename}`);
        console.log(`   📍 ${g.scenario}`);
        console.log(`   📝 ${g.copy}`);
    }

    return generated;
}

// Parse args
function parseArgs() {
    const args = process.argv.slice(2);
    const opts = { profile: null, category: null, count: 1, aspect: '1:1', resolution: '2K' };

    for (let i = 0; i < args.length; i++) {
        if (['-p', '--profile'].includes(args[i])) opts.profile = args[++i];
        if (['-c', '--category'].includes(args[i])) opts.category = args[++i];
        if (['-n', '--count'].includes(args[i])) opts.count = parseInt(args[++i]) || 1;
        if (['-a', '--aspect'].includes(args[i])) opts.aspect = args[++i];
        if (['-r', '--resolution'].includes(args[i])) opts.resolution = args[++i];
        if (['-h', '--help'].includes(args[i])) {
            console.log(`
Sistema iGaming - Gemini 3 Pro Image

Uso: node igaming-gemini.js -p PERFIL [opções]

  -p  Perfil: teo, jonathan, pedro, laise, fernandes
  -c  Categoria: viagem, praia, academia, restaurante, trabalho, lifestyle, cassino
  -n  Quantidade (default: 1)
  -a  Aspecto: 1:1, 16:9, 9:16 (default: 1:1)
  -r  Resolução: 1K, 2K, 4K (default: 2K)

Exemplos:
  node igaming-gemini.js -p teo -c praia -n 3
  node igaming-gemini.js -p jonathan -c viagem -n 5 -a 16:9
`);
            process.exit(0);
        }
    }
    return opts;
}

// Main
async function main() {
    const opts = parseArgs();

    if (!opts.profile) {
        console.log('❌ Perfil obrigatório! Use -p teo');
        process.exit(1);
    }

    console.log('🎮 SISTEMA IGAMING - GEMINI 3 PRO IMAGE');
    console.log('='.repeat(50));

    await generateForProfile(opts.profile, opts.category, opts.count, opts.aspect, opts.resolution);
}

main().catch(console.error);
